### ISSUE TITLE
OnCameraChange method causes crash

### ISSUE THREAD

**Issue by vladislavsosiuk** [amay077/Xamarin.Forms.GoogleMaps#352]
(This is template for issue. Please choose from feature request or bug report or others then write.
If you need realtime communication then you can join our [gitter room](https://gitter.im/Xamarin-Forms-GoogleMaps/public).)

----
### SUMMARY

(write summary of your feature request or suggestion)

### DETAILS

Xamarin caused by: android.runtime.JavaProxyThrowable: System.NullReferenceException: Object reference not set to an instance of an object.
Xamarin.Forms.GoogleMaps.Android.MapRenderer.OnCameraChange(CameraPosition pos)<6fc4378ed98443b1ab50dfb0cc46da2e>:0
Android.Gms.Maps.GoogleMap.IOnCameraChangeListenerInvoker.n_OnCameraChange_Lcom_google_android_gms_maps_model_CameraPosition_(IntPtr jnienv, IntPtr native__this, IntPtr native_position)<edd902d5391542adb87c92c0219e00cb>:0
at (wrapper dynamic-method) System.Object:c9033d22-a901-4c35-b3fa-3c6ed402cb2b (intptr,intptr,intptr)
md58c1d70bdc76f7066f10e739f3153fe8d.MapRenderer.n_onCameraChange(Native Method)
md58c1d70bdc76f7066f10e739f3153fe8d.MapRenderer.onCameraChange()MapRenderer.java:61
com.google.android.gms.maps.GoogleMap$18.onCameraChange(Unknown Source)
com.google.android.gms.maps.internal.zze$zza.onTransact(Unknown Source)
android.os.Binder.transact()Binder.java:380
at com.google.android.gms.maps.internal.x.a(:com.google.android.gms.DynamiteModulesB:12)
at com.google.maps.api.android.lib6.gmm6.api.b.run(:com.google.android.gms.DynamiteModulesB:12)
android.os.Handler.handleCallback()Handler.java:739
android.os.Handler.dispatchMessage()Handler.java:95
android.os.Looper.loop()Looper.java:145
android.app.ActivityThread.main()ActivityThread.java:5951
java.lang.reflect.Method.invoke(Native Method)
java.lang.reflect.Method.invoke()Method.java:372
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run()ZygoteInit.java:1400
com.android.internal.os.ZygoteInit.main()ZygoteInit.java:1195

```
I call moveToPosition method in OnAppearing method of my page, if press back button after map will be with white color background, but before map will appear it causes such crash
```

### PLATFORMS

Android

- [x] Android
- [ ] iOS
- [ ] UWP

----
### SUMMARY

(write summary of your bug report.)

### VERSIONS

* Xamarin.Forms.GoogleMaps - 2.1.0
* Xamarin.Forms - 2.3.4.247
* 

### PLATFORMS

- [ ] Android
- [] iOS
- [ ] UWP

### HOW TO REPRODUCE

(Let me show how to reproduce with steps or codes)

1. 
2. 
3. 

```csharp
// code for reproduce
```

### ACTUAL IMAGES

(Let me show your actual problem images/stacktrace. If you want to capture simulator screen to animation GIF, I recomend to [LICEcap](http://www.cockos.com/licecap/).)

![something_image](your_image.png) 

```
// paste stacktrace
```

----

## OTHERS

(free format)

**Comment by amay077** [comment]
Could you show us how to reproduce?

**Comment by vladislavsosiuk** [comment]
Here is link for video which shows needed behavior

https://www.youtube.com/watch?v=E8viGkj6lyc


пн, 24 июля 2017 г. в 20:43, amay077 <notifications@github.com>:

**Comment by vladislavsosiuk** [comment]
So sorry, wrong link...

https://drive.google.com/open?id=0B8zdMoSojSe9ejlyMTJ4aEhmTkE

пн, 24 июля 2017 г. в 20:51, Владислав Сосюк <vladislavsosiuk@gmail.com>:

**Comment by vladislavsosiuk** [comment]
Here is screen shot of hockeyapp crash report

**Comment by amay077** [comment]
Sorry, can you make a small project for reproduce it?
I can not debug your "product" project.

**Comment by vladislavsosiuk** [comment]
sorry, don’t have time now for creating other projects, but here is something useful for you:
I have copied your source code and caught the bug: Map property is null at the moment when it calls Map.CameraPosition = camera; so maybe just use usual try catch operator in new version inside of this method

**Comment by amay077** [comment]
👍 
I hope you report detail to me if you can take the time.

**Comment by urnfield0** [comment]
Hi,

I can repeat this (or a very similar exception) pretty reliably on Android by navigating to any Forms page that contains a Map, and then hitting the back button (either in the top navigation bar, or the Android back button) immediately, before the page has been displayed.

I'm using Xamarin.Forms.GoogleMaps version 2.1.0, the full exception stack, as captured by HockeyApp, is below.

Let me know if there's any other info that would be useful.

`Xamarin caused by: android.runtime.JavaProxyThrowable: System.NullReferenceException: Object reference not set to an instance of an object.
Xamarin.Forms.GoogleMaps.Android.MapRenderer.OnCameraChange(CameraPosition pos)<33f480a5aa3448b39a955c15f6587f42>:0
Android.Gms.Maps.GoogleMap.IOnCameraChangeListenerInvoker.n_OnCameraChange_Lcom_google_android_gms_maps_model_CameraPosition_(IntPtr jnienv, IntPtr native__this, IntPtr native_position)<edd902d5391542adb87c92c0219e00cb>:0
at (wrapper dynamic-method) System.Object:30647b8f-d451-41cb-acbf-18004240faf3 (intptr,intptr,intptr)
md59e1546199385abe7b23a89990a0f94aa.MapRenderer.n_onCameraChange(Native Method)
md59e1546199385abe7b23a89990a0f94aa.MapRenderer.onCameraChange()MapRenderer.java:61
com.google.android.gms.maps.GoogleMap$18.onCameraChange()Unknown Source:2
com.google.android.gms.maps.internal.zze$zza.onTransact()Unknown Source:35
android.os.Binder.transact()Binder.java:604
at gl.b(:com.google.android.gms.DynamiteModulesB@11509940:20)
at com.google.android.gms.maps.internal.z.a(:com.google.android.gms.DynamiteModulesB@11509940:5)
at com.google.maps.api.android.lib6.gmm6.api.b.run(:com.google.android.gms.DynamiteModulesB@11509940:12)
android.os.Handler.handleCallback()Handler.java:789
android.os.Handler.dispatchMessage()Handler.java:98
android.os.Looper.loop()Looper.java:164
android.app.ActivityThread.main()ActivityThread.java:6541
java.lang.reflect.Method.invoke(Native Method)
com.android.internal.os.Zygote$MethodAndArgsCaller.run()Zygote.java:240
com.android.internal.os.ZygoteInit.main()ZygoteInit.java:767`

**Comment by amay077** [comment]
Thanks!
I reproduced it.

**Comment by amay077** [comment]
Fixed in [v2.1.1-beta2](https://github.com/amay077/Xamarin.Forms.GoogleMaps/releases/tag/v2.1.1-beta2) .

**Comment by MisterJimson** [comment]
Slightly different stack but same issue, on version 2.1.1.

Reproduced by loading the map and very quickly navigating away.

[MonoDroid] UNHANDLED EXCEPTION:
[MonoDroid] System.NullReferenceException: Object reference not set to an instance of an object.
[MonoDroid]   at Xamarin.Forms.GoogleMaps.Logics.BaseCameraLogic`1[TNativeMap].Register (Xamarin.Forms.GoogleMaps.Map map, TNativeMap nativeMap) [0x00021] in <85b6a1e995414b49807a5823c057f27f>:0 
[MonoDroid]   at Xamarin.Forms.GoogleMaps.Android.MapRenderer+<OnElementChanged>d__20.MoveNext () [0x00339] in <befa768c7be345e0b5536e31ca07115f>:0 
[MonoDroid] --- End of stack trace from previous location where exception was thrown ---
[MonoDroid]   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <896ad1d315ca4ba7b117efb8dacaedcf>:0 
[MonoDroid]   at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__6_0 (System.Object state) [0x00000] in <896ad1d315ca4ba7b117efb8dacaedcf>:0 
[MonoDroid]   at Android.App.SyncContext+<>c__DisplayClass2_0.<Post>b__0 () [0x00000] in /Users/builder/data/lanes/5147/7a6a056e/source/xamarin-android/src/Mono.Android/Android.App/SyncContext.cs:35 
[MonoDroid]   at Java.Lang.Thread+RunnableImplementor.Run () [0x00008] in /Users/builder/data/lanes/5147/7a6a056e/source/xamarin-android/src/Mono.Android/Java.Lang/Thread.cs:36 
[MonoDroid]   at Java.Lang.IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this) [0x00008] in /Users/builder/data/lanes/5147/7a6a056e/source/xamarin-android/src/Mono.Android/obj/Release/android-25/mcw/Java.Lang.IRunnable.cs:81 
[MonoDroid]   at (wrapper dynamic-method) System.Object:3fb91e7e-e7b9-48a1-9627-926b0fb4764b (intptr,intptr)
